qutip/core/data/expm.py

```diff
@@ -58,6 +66,28 @@ def expm_csr_dense(matrix: CSR) -> Dense:
 expm.add_specialisations([
     (CSR, CSR, expm_csr),
     (CSR, Dense, expm_csr_dense),
+    (Dense, Dense, expm_dense),
```

Comment:
What did v5 do for Dense .expm before this?

Reply:
It would convert it to `CSR` automatically and use `expm_csr_dense`. The Dispatcher automatically fill missing specializations.

Reply:
I suspected as much -- good to fix that.